pytorch_lightning/utilities/cli.py

```diff
@@ -37,13 +37,17 @@
 
 if _JSONARGPARSE_AVAILABLE:
     from jsonargparse import ActionConfigFile, ArgumentParser, class_from_function, Namespace, set_config_read_mode
-    from jsonargparse.optionals import import_docstring_parse
 
     set_config_read_mode(fsspec_enabled=True)
 else:
     locals()["ArgumentParser"] = object
     locals()["Namespace"] = object
 
+try:
+    import docstring_parser
+except ImportError:
+    docstring_parser = False
+
```

Comment:
Oh, mypy is (rightfully) complaining also about this. Can we have some guarding variable so that we are not assigning bool to module?